overloads/matrix.h

```diff
@@ -0,0 +1,49 @@
+/*
+    This file defines the various mathematical overloads of the Matrix Class.
+*/
+
+#include <types/matrix.h>
+
+#ifndef __MATRIX_OPS_INCLUDED__   
+#define __MATRIX_OPS_INCLUDED__  
+
+// Sigmoid 
+namespace matrixOps {
+    template<typename T>
+    Matrix<T> sigmoid(const Matrix<T> &a) {
+        return (T)1/((T)1 + (((T)-1)*a).exp());
+    }
+};
+
+// Overloaded function for printing matrix: cout<<matrix<<endl;
+template<typename T>
+ostream & operator << (ostream &out, Matrix<T> &m) {
+    vector<int> stack;
+    return m.print(out,stack,0);
+}
+
+// Divison with a scalar as divident
+template<typename E>
```

Comment:
nit: Any particular reason not for using 'T' in the typename arg here?

Reply:
Yes, fixed. Changed to T everywhere.